test/e2e2/conftest.py

```diff
@@ -413,12 +413,18 @@ def faketime_get():
         return '+0d'
 
 
-def faketime_add(days: int):
-    assert days >= 0
+class FaketimeShift:
+    def __init__(self, days: int):
+        assert days >= 0
+        self.days = days
 
-    current_faketime = faketime_get()
-    assert current_faketime[0] == '+'
-    assert current_faketime[-1] == 'd'
-    current_days = int(current_faketime[1:-1])
+    def __enter__(self):
+        self._faketime = faketime_get()
+        assert self._faketime[0] == '+'
+        assert self._faketime[-1] == 'd'
+        current_days = int(self._faketime[1:-1])
 
-    faketime(f'+{current_days + days:n}d')
+        faketime(f'+{current_days + self.days:n}d')
+
+    def __exit__(self, type, value, traceback):
+        faketime(self._faketime)
```

Comment:
Going back in time may confuse other components of desec-stack and could lead to *weird* seemingly nonsensical errors of subsequent tests. In other words, going back in time could have unforeseen consequences on subsequent tests, ruining the test independence. On the other hand, I can clearly see the advantage in resetting the clock to original time. So, no strong opinion here on either option, just pointing out a potential trap.

Reply:
Fair enough. I'd say, let's see if it causes problems :)

Reply:
Alright let's add a usage warning (code comment) to the context manager

Reply:
done